### ISSUE TITLE
Code refactor

### ISSUE THREAD

**Issue by groupsky** [atom-community/sync-settings#79]
Refactor the core logic for backing and restoring to remove the duplicated code and logic. A possible solution would be something like:

``` coffeescript
# define the configuration files
fileConfig = 
  'settings.json': {reader: => JSON.stringify(atom.config.settings, @filterSettings, '\t'), writer: (content) => @applySettings '', JSON.parse(content)}

# backup
files = for file, config of fileConfig
  config.reader() if atom.config.get("sync-settings.sync#{_.capitalize file}")

# restore
for own filename, file of res.files
  if filename of fileConfig
    fileConfig[filename].writer(file.content) if atom.config.get("sync-settings.sync#{_.capitalize file}")
```

warning: haven't tested the code, so it may need some tuning

**Comment by jerone** [comment]
I like your idea of `reader` & `writer`. Thinking of separating more parts into classes and this is a good example.

**Comment by jerone** [comment]
Some code to look into:
* https://github.com/atom/metrics/blob/master/lib/reporter.coffee#L52-L61

**Comment by jerone** [comment]
Done some basic ground work for refactoring: https://github.com/jerone/atom-sync-settings/tree/issues/%2379-refactor
* move some sync type to it's own file.
* setup interface from which all sync types should extend from.
* let the sync manager read the `sync-interfaces` folder and list all sync interfaces.

**Comment by jerone** [comment]
I'll include some logic to base64 encode images into the refactoring to fix #93.

Any other non-text and non-image files we should block.
